packages/select/src/elements/SelectField.js

```diff
@@ -52,11 +52,21 @@ export default class SelectField extends ControlledComponent {
     };
   };
 
+  applyMarginStyling = otherProps => ({ style: { marginTop: '8px' }, ...otherProps });
```

Comment:
This does not capture the nuance provided by https://github.com/zendeskgarden/css-components/blob/master/packages/forms/src/_text/_label.css which accounts for `--sm` sized fields, handles `--inline`, and special-casing for `.c-txt__label` followed by a `.c-txt__hint`.

Although it's not generally advisable, I think we should take a serious look at [render-fragment](https://www.npmjs.com/package/render-fragment) in order to provide a first-class styling experience for React 16 (allowing React 15 users to manipulate the layout in a more manual fashion). That approach would get us through this somewhat hacky `has[?]Rendered` code in the short-to-medium term.

Reply:
Even though `render-fragment` will create different DOM based on the version, I think you're right that it's going to be the easiest way forward.

I'll make sure to include a warning in the documentation as well for this with example of correct styling. The good news is that consumers can continue to style this with fela/css-modules however they want to get the desired layout.